base/operators.jl

```diff
@@ -1339,8 +1339,7 @@ struct Splat{F} <: Function
     Splat(f) = new{Core.Typeof(f)}(f)
 end
 (s::Splat)(args) = s.f(args...)
-print(io::IO, s::Splat) = print(io, "splat(", s.f, ')')
-show(io::IO, s::Splat) = print(io, s)
+show(io::IO, s::Splat) = print(io, "splat(", s.f, ')')
```

Comment:
also a related bugfix:
```suggestion
show(io::IO, s::Splat) = (print(io, "splat("); show(io, s.f); print(io, ")"))
```